examples/Fletcher_random_stream_performance/Fletcher_random_stream_performance.ino

```diff
@@ -16,166 +16,214 @@
 #else
 #define MAX_LEN 16384
 #endif
+union main_value_storage {
+  uint8_t uint8[MAX_LEN];
+  uint16_t uint16[MAX_LEN / 2];
+  uint32_t uint32[MAX_LEN / 4];
+} values;
 
 #define DO_N 23
 
-uint32_t myfletcher16(uint8_t *data, const uint16_t length)
-{
-  uint16_t s1 = 0;
-  uint16_t s2 = 0;
-  for (uint16_t i = 0; i < length; i++)
-  {
-    s1 = (s1 + data[i]) % ((((uint16_t) 1) << 8) - 1);
-    s2 = (s2 + s1) % ((((uint16_t) 1) << 8) - 1);
+#if defined(ARDUINO_ARCH_AVR)
+String float2strn(float value, size_t n) {
+  char strvalout[n];
+  dtostrf(value, n, 2, strvalout);
+  return strvalout;
+}
+#else
+String float2strn(float value, size_t n) {
+  String strval = String(value);
+  while (strval.length() < n) {
+    strval = " " + strval;
   }
-  return (s2 << 8) | s1;
+  return strval;
 }
+#endif
 
-uint32_t myfletcher32(uint16_t *data, const uint16_t length)
+uint16_t myfletcher16(uint8_t *data, const size_t length)
 {
-  uint32_t s1 = 0;
-  uint32_t s2 = 0;
-  for (uint16_t i = 0; i < length; i++)
+  uint8_t s1 = 0;
+  uint8_t s2 = 0;
+  for (size_t i = 0; i < length; i++)
   {
-    s1 = (s1 + data[i]) % ((((uint32_t) 1) << 16) - 1);
-    s2 = (s2 + s1) % ((((uint32_t) 1) << 16) - 1);
+    s1 = (uint8_t) (((uint16_t) s1 + (uint16_t) data[i]) % ((((uint16_t) 1) << 8) - ((uint16_t) 1)));
```

Comment:
Still looks far from the reference as described on Wikipedia. 
No need to change it, it is a matter of taste.

Reply:
OK, it's hard to read. A changed it to:

```c++
 // UINT8_MAX = 255 = ((((uint16_t) 1) << 8) - ((uint16_t) 1))
    s1 = (uint8_t) (((uint16_t) s1 + (uint16_t) data[i]) % UINT8_MAX);
    s2 = (uint8_t) (((uint16_t) s2 + (uint16_t) s1) % UINT8_MAX);
```

If you forget all the castings, it's identical to the one on [wikipedia: Fletcher's checksum](https://en.wikipedia.org/wiki/Fletcher%27s_checksum#Straightforward):

```c++
    s1 = (s1 + data[i]) % UINT8_MAX;
    s2 = (s2 + s1) % UINT8_MAX;
```

Reply:
I know it is the same, as said before it is a matter of taste.